routers/web/repo/actions/actions.go

```diff
@@ -107,7 +107,12 @@ func List(ctx *context.Context) {
 			// The workflow must contain at least one job without "needs". Otherwise, a deadlock will occur and no jobs will be able to run.
 			hasJobWithoutNeeds := false
 			// Check whether have matching runner and a job without "needs"
+			emptyJobsNumber := 0
```

Comment:
Do we really need this? If no job in Jobs, then we will show `no_job_without_needs` error.

Reply:
It's necessary because .ie.

```yaml
on: [push]

jobs:
  jobs:
```

Reply:
This PR LGTM as a quick fix.

And I also agree refactor it later, move `no_job_without_needs` and `no_job` to `ReadWorkflow`, the error is just `invalid_workflow`.